state/machine_linklayerdevices.go

```diff
@@ -1120,3 +1120,95 @@ func generateMACAddress() string {
 	}
 	return fmt.Sprintf(macAddressTemplate, digits...)
 }
+
+type MachineNetworkInfoResult struct {
+	NetworkInfos []network.NetworkInfo
+	Error        *error
+}
+
+// Add address to a device in list or create a new device with this address
+func addAddressToResult(networkInfos []network.NetworkInfo, address *Address) []network.NetworkInfo {
+	ifaceAddress := network.InterfaceAddress{
+		Address: address.Value(),
+		CIDR:    address.SubnetCIDR(),
+	}
+	for i := range networkInfos {
+		networkInfo := &networkInfos[i]
+		if networkInfo.InterfaceName == address.DeviceName() {
```

Comment:
Interface name for vlan device is different than for vlan0 device - it has ".vlanid" appended.